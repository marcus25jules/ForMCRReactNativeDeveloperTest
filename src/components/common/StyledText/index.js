import React from "react";
import PropTypes from "prop-types";
import { Text } from "react-native";
import { moderateScale } from "utils/scale";

const StyledText = ({
  size = 14,
  color,
  weight,
  style = {},
  fontFamily = "Myriad Pro",
  testID,
  accessibilityLabel,
  ...rest
}) => {
  return (
    <Text
      {...rest}
      testID={testID}
      accessibilityLabel={accessibilityLabel}
      allowFontScaling={false}
      style={[
        {
          fontSize: moderateScale(size),
          color,
          fontWeight: weight,
          flexWrap: "wrap",
          fontFamily: fontFamily,
        },
        style,
      ]}
    />
  );
};

StyledText.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  weight: PropTypes.string
};

StyledText.defaultProps = {
  size: 14,
  color: "#000000",
  weight: "normal",
  flexWrap: "wrap",
};

export default StyledText;
