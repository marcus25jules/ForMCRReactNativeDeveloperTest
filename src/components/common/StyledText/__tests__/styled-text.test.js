import React from "react";
import { render } from "@testing-library/react-native";

import StyledText from "../index";

/**
  Passed Rendering component
**/

describe("Test StyledText Component", () => {
  it("renders correctly", () => {
     const { toJSON } = render(
       <StyledText
         color={"black"}
         weight={"normal"}
         testID="test-styled-text"
         accessibilityLabel="test-styled-text"
       />,
     );
     expect(toJSON).toMatchSnapshot();
  });
});
