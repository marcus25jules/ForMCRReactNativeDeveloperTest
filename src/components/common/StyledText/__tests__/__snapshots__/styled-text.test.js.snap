// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test StyledText Component renders correctly 1`] = `[Function]`;
